app/helpers/application_helper/button/basic.rb

```diff
@@ -0,0 +1,25 @@
+class ApplicationHelper::Button::Basic < Hash
+  def initialize(view_context, view_binding, instance_data, props)
+    @view_context  = view_context
+    @view_binding  = view_binding
+
+    props.each { |k, v| self[k] = v }
+
+    instance_data.each do |name, value|
+      instance_variable_set(:"@#{name}", value)
+    end
+  end
+
+  def calculate_properties
+    props["enabled"] = "false" if disabled?
```

Comment:
I wasn't actually referring to `"enabled"` vs `:enabled`, but to `"false"` vs `false` ;)